cluster-autoscaler/core/scale_down.go

```diff
@@ -103,6 +102,112 @@ func (n *NodeDeleteStatus) SetDeleteInProgress(status bool) {
 	n.deleteInProgress = status
 }
 
+type resourceName string
+type scaleDownResourcesLimits map[resourceName]int64
+type scaleDownResourcesDelta map[resourceName]int64
+
+// used as a value in scaleDownResourcesLimits if actual limit could not be obtained due to errors talking to cloud provider
+const limitUnknown = math.MinInt64
+
+func computeScaleDownResourcesLeftLimits(nodes []*apiv1.Node, resourceLimiter *cloudprovider.ResourceLimiter, timestamp time.Time) scaleDownResourcesLimits {
+	totals := make(map[resourceName]int64)
+	totals[cloudprovider.ResourceNameCores], totals[cloudprovider.ResourceNameMemory] = calculateCoresAndMemoryTotal(nodes, timestamp)
+	resultScaleDownLimits := make(scaleDownResourcesLimits)
+
+	for _, resource := range resourceLimiter.GetResources() {
```

Comment:
I still think it doesn't make sense to recompute the map we already have here (see discussion at https://github.com/kubernetes/autoscaler/pull/926/files#r193725222 .) It adds unnecessary ~30 lines of code (plus 1 file, if counting tests) and complicates rather than simplifies the logic. It's probably otherwise harmless, though.

Reply:
Just for sake of not going back and forth with that I will stay with this shape. And as we finish scale_up let's take a look how it looks like and if look redundant then I will prepare a cleanup commit. Ok?

Reply:
Let's leave it as it is if you prefer, I just disagree:) We can probably revisit it if we ever refactor ResourceLimiter